qemu/tests/blockdev_snapshot_nonexist_target.py

```diff
@@ -0,0 +1,32 @@
+import re
+
+from virttest.qemu_monitor import QMPCmdError
+
+from provider.blockdev_snapshot_base import BlockDevSnapshotTest
+
+
+class BlockdevSnapshotNonexistTarget(BlockDevSnapshotTest):
+    def run_test(self):
+        if not self.main_vm.is_alive():
+            self.main_vm.create()
+        self.main_vm.verify_alive()
+        try:
+            self.create_snapshot()
+        except QMPCmdError as e:
+            qmp_error_msg = self.params.get("qmp_error_msg")
+            if not re.search(qmp_error_msg, str(e.data)):
+                self.test.fail(str(e))
+
```

Comment:
we may also need to add 'else' after 'except', in case create_snapshot succeeds unexpectedly

Reply:
And I guess that create_snapshot may succeed in this case, for the snapshot file has been created successfully. 
we may need to re-define prepare_snapshot_file as:
def prepare_snapshot_file():
pass

Reply:
Have added 'else', please help to check it, zhenchao.

And we don't run prepare_snapshot_file() here, I re-define def run_test(self) here, which just start vm and do snapshot to sn1 (not created).

Reply:
LGTM